nonempty.nix

```diff
@@ -0,0 +1,461 @@
+with rec {
+  function = import ./function.nix;
+  inherit (function) const flip compose id;
+
+  num = import ./num.nix;
+  inherit (num) min max;
+
+  list = import ./list.nix;
+  optional = import ./optional.nix;
+};
+
```

Comment:
add comment

/* type nonempty a = { head :: a, tail :: [a] }
*/

Reply:
added in f2ff276106a579fd984d1e42cbe190d956b7b63b